source/4.0/learn/introduction/active-record.html.md

```diff
@@ -1,82 +1,545 @@
+  def first_name
+    name.split(' ').first
+  end
 
-There is no single "model" object in ROM. ROM objects are instantiated by
-the mappers and have no knowledge about persistence. You can map to whatever
-structure you want and in common use-cases you can use repositories to
-automatically map query results to simple struct-like objects.
+  def last_name
+    name.split(' ').last
+  end
+end
+
+user = User.first
+#> #<User id: 1, name: "Jane Doe">
+
+user.first_name
+#> "Jane"
+
+user.last_name
+#> "Doe"
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+class Users < ROM::Relation[:sql]
+  struct_namespace Entities
+
+  schema(infer: true)
+end
+
+module Entities
+  class Users < ROM::Struct
+    def first_name
+      name.split(' ').first
+    end
+
+    def last_name
+      name.split(' ').last
+    end
+  end
+end
+
+user = users_relation.first
+#> #<Entitites::User id=1 name="Jane Doe">
+
+user.first_name
+#> "Jane"
+
+user.last_name
+#> "Doe"
+```
+
+For a brief overview and links to more in-depth information about relations see
+the Relations section in our
+[Core Concepts](/%{version}/learn/getting-started/core-concepts#relations)
+guide.
+
+## Queries
+
+### Basic Queries
+
+Once you have a relation, it becomes almost trivial to start querying for
+information in a similar fashion as ActiveRecord. A basic example below:
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User.where(name: "Jane").first
+#> #<User id: 1, name: "Jane">
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation.where(name: "Jane").first
+#<ROM::Struct::User id=1 name="Jane">
+```
+
+### Query Subset of Data
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User
+  .select("name")
+  .where(name: name)
+  .first
+
+#> #<Users id: nil, name: "Jane">
+#
+# Little bit later... 🔥🔥 BOOM! 🔥🔥
+# ActiveModel::MissingAttributeError
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation
+  .select(:name)
+  .where(name: name)
+  .one
+
+#> #<ROM::Struct::User name="Jane">
+```
+
+When designing an ActiveRecord model the developer needs to take into account
+every different way that a model will be used in every context. This means
+either loading too much data that isn't needed to satisfy strict validations or
+making those validations so vague as to allow for different data configurations
+rendering them effectively meaningless forcing the need to test the model every
+time before use.
+
+Since it's so easy with ROM to load only pertinent data, we focus on allowing
+the user to map data into their desired format. Then using context aware
+validations at the system boundaries, developers can trust the data is correct
+and utilize it without fear.
+
+### Query with Complex Conditions
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User.where("admin IS ? OR moderator IS ?", true, true)
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation.where { admin.is(true).or(moderator.is(true)) }
```

Comment:
there's also `|`, `where { admin.is(true) | moderator.is(true) }`

Reply:
actually, there IS NO `or` (yet) :/

Reply:
Huh.. I swear I tested that code sample. Ah well, must be crazy